File: Numan Kafadar/week2/numberofrecentcalls.cpp

class RecentCounter {
    int ct;
public:
    queue<int> q;
    RecentCounter() {
        ct=0;
    }
    
    int ping(int t) {
        if(q.empty())
        q.push(t);
        else{
            q.push(t);
            if(q.front()<t-3000 || q.front()>t){
            while((q.front()<t-3000 || q.front()>t)&&!q.empty())
                q.pop();
            }
        }
        return ct=q.size();
    }
};

File: Galip Tezcan/2. Hafta/number_of_recent_calls.py
class queue:
    def __init__(self):
        self.items = []

    def isEmpty(self):
        return self.items == []

    def push(self, item):
         self.items.append(item)

    def pop(self):
        if self.items != []:
            return self.items.pop(0)

    def peek(self):
        if self.items != []:
            return self.items[0]

    def size(self):
        return len(self.items)




class RecentCounter:

    def __init__(self):
        self.queue_list = queue()

    def ping(self, t: int):
        self.queue_list.push(t)

        while self.queue_list.peek() < t - 3000:
            self.queue_list.pop()

        return self.queue_list.size()